compiler+runtime/src/cpp/jank/runtime/obj/persistent_string.cpp

```diff
@@ -62,6 +63,41 @@ namespace jank::runtime::obj
     return data.compare(s.data);
   }
 
+  object_ptr persistent_string::get(object_ptr const key) const
+  {
+    return get(key, nil::nil_const());
+  }
+
+  object_ptr persistent_string::get(object_ptr const key, object_ptr const fallback) const
+  {
+    if(key->type == object_type::integer)
+    {
+      auto const i(static_cast<size_t>(expect_object<integer>(key)->data));
+      if(data.size() <= i)
+      {
+        return nil::nil_const();
+      }
+      return make_box<runtime::obj::character>(data[i]);
+    }
+    return fallback;
+  }
+
+  native_bool persistent_string::contains(object_ptr const key) const
+  {
+    if(key->type == object_type::integer)
+    {
+      auto const i(static_cast<size_t>(expect_object<integer>(key)->data));
+      return 0 <= i && i < data.size();
+    }
+    throw std::runtime_error{ fmt::format("contains? not supported on string: {}",
+                                          runtime::to_string(key)) };
```

Comment:
This is a confusing error message. It says to me that the thing which comes after `:` is a string which doesn't support `contains?`.

Reply:
Returns `false` instead of throwing now.